### PR TITLE
[leica/5.4-2.1.x-imx/crocodile]: battery and magnetometer (dts and config)

### PR DESCRIPTION
**Pull request by Qingwu-Li**
Crocodile device tree and configure file:
Add battery fuel-gauge bq78z100.
Add magnetometer iis2mdc.
Both are on I2C3 bus.

### CODE REVIEW

File: arch/arm64/boot/dts/freescale/crocodile.dts [hunk 1]
```diff
@@ -181,6 +181,17 @@
 		model = "imx-audio-micfil";
 		cpu-dai = <&micfil>;
 	};
+	bat: battery {
```
Comment (zandrey) [inline arch/arm64/boot/dts/freescale/crocodile.dts:184]: Not sure if we need a label for the battery node, even if it is listed in the documentation as an example...

Can you please use the simple naming scheme here for the node, like:

```
internal-battery {
	<content>
}
```

**Review by zandrey — CHANGES_REQUESTED** [review]

**Comment by Qingwu-Li** [comment]
Thanks you @zandrey, I reworked according your comments.

File: arch/arm64/boot/dts/freescale/crocodile.dts (hunk 1 above)
Comment (Qingwu-Li) [reply arch/arm64/boot/dts/freescale/crocodile.dts]: @zandrey ,the lable was used by node "bq78z100: fuel-gauge@55" maybe we need it.

**Comment by Qingwu-Li** [comment]
@zandrey I add the bat lable again, since it was used by bq78z100

File: arch/arm64/boot/dts/freescale/crocodile.dts [hunk 2]
```diff
@@ -181,6 +181,17 @@
 		model = "imx-audio-micfil";
 		cpu-dai = <&micfil>;
 	};
+	bat: Internal-battery {
```
Comment (zandrey) [inline arch/arm64/boot/dts/freescale/crocodile.dts:184]: Please avoid using capital letters in node names, use `internal-battery` instead.

**Review by zandrey — CHANGES_REQUESTED** [review]

**Review by zandrey — APPROVED** [review]

**Comment by Qingwu-Li** [comment]
Thank you @zandrey

**Pull request merged**